README.md
# install

```
yarn add wxemoji-awesome
```

# Usage

```js
import "wxemoji-awesome/lib/index.css";
```

```html
<img class="wxem wxem-wx-1" alt="[微笑]" />
```


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.0.11](https://github.com/planjs/wxemoji-awesome/compare/stable-0.0.10...stable-0.0.11) (2023-12-27)

### [0.0.10](https://github.com/planjs/wxemoji-awesome/compare/stable-0.0.9...stable-0.0.10) (2023-12-27)


### Bug Fixes

* update logic ([cae1b81](https://github.com/planjs/wxemoji-awesome/commit/cae1b811ce21744c09eeeef7f0b054237c1531bc))
* update MD ([155b980](https://github.com/planjs/wxemoji-awesome/commit/155b9803214fe06f389115189eeca3f141f4b786))

### 0.0.9 (2023-12-27)


### Bug Fixes

* class name ([1d55e38](https://github.com/planjs/wxemoji-awesome/commit/1d55e380c11fab1d3fb4ac58cec1456b32c3d705))
* doc ([2741fff](https://github.com/planjs/wxemoji-awesome/commit/2741fff9ffa04467e4dc1ce3fce09033f833b4b4))
* img url ([ec845c1](https://github.com/planjs/wxemoji-awesome/commit/ec845c14ed166e2d558d1bc0180413550166ab69))
* remove deps ([df45a87](https://github.com/planjs/wxemoji-awesome/commit/df45a87930ff734c8f1a0a5ea40193cbe98a97e4))
* types ([2f6835d](https://github.com/planjs/wxemoji-awesome/commit/2f6835db74766a1b11b8f22d6990e16aec149ecf))
* wxemoji-awesome ([2932f7a](https://github.com/planjs/wxemoji-awesome/commit/2932f7a88459758e603cca422c05ff893daec797))

### 0.0.8 (2020-11-26)


### Bug Fixes

* class name ([1d55e38](https://github.com/planjs/wxemoji-awesome/commit/1d55e380c11fab1d3fb4ac58cec1456b32c3d705))
* doc ([2741fff](https://github.com/planjs/wxemoji-awesome/commit/2741fff9ffa04467e4dc1ce3fce09033f833b4b4))
* img url ([ec845c1](https://github.com/planjs/wxemoji-awesome/commit/ec845c14ed166e2d558d1bc0180413550166ab69))
* remove deps ([df45a87](https://github.com/planjs/wxemoji-awesome/commit/df45a87930ff734c8f1a0a5ea40193cbe98a97e4))
* types ([2f6835d](https://github.com/planjs/wxemoji-awesome/commit/2f6835db74766a1b11b8f22d6990e16aec149ecf))
* wxemoji-awesome ([2932f7a](https://github.com/planjs/wxemoji-awesome/commit/2932f7a88459758e603cca422c05ff893daec797))


src/utils.js
const fs = require('fs');
const path = require('path');

async function deleteFile(path) {
  const exists = fs.existsSync(path);
  if (!exists) {
    return fs.mkdirSync(path);
  }
  var files = [];
  if (fs.existsSync(path)) {
    files = fs.readdirSync(path);
    files.forEach(function (file) {
      var curPath = path + '/' + file;
      if (fs.statSync(curPath).isDirectory()) {
        deleteFile(curPath);
      } else {
        fs.unlinkSync(curPath);
      }
    });
  }
}

function copyFolder(srcDir, tarDir) {
  function copyFile(srcPath, tarPath) {
    var rs = fs.createReadStream(srcPath);
    var ws = fs.createWriteStream(tarPath);
    rs.pipe(ws);
  }

  fs.readdir(srcDir, function (err, files) {
    if (err) {
      return;
    }

    files.forEach(function (file) {
      var srcPath = path.join(srcDir, file);
      var tarPath = path.join(tarDir, file);

      fs.stat(srcPath, function (err, stats) {
        if (err) {
          return;
        }
        if (stats.isDirectory()) {
          console.log('mkdir', tarPath);
          fs.mkdir(tarPath, function (err) {
            if (err) {
              console.log(err);
              return;
            }

            copyFolder(srcPath, tarPath);
          });
        } else {
          copyFile(srcPath, tarPath);
        }
      });
    });
  });
}

module.exports = {
  deleteFile,
  copyFolder
}

src/update.js
const request = require('request');
const fs = require('fs');
const path = require('path');
const { deleteFile, copyFolder } = require('./utils')

function fetchWxEmoji(opts = {}, path = '') {
  return new Promise((resolve, reject) => {
    request.head(opts.url, (err, res, body) => {
      if (err) {
        return reject(err);
      } else {
        if (res.statusCode !== 200) {
          // eslint-disable-next-line prefer-promise-reject-errors
          return reject(`${opts.url}:${res.statusMessage}`);
        }

        request
          .get(opts)
          .pipe(fs.createWriteStream(path))
          .on('error', e => {
            resolve(e);
          })
          .on('finish', () => {
            resolve({
              res,
              body,
            });
          });
      }
    });
  });
}

async function update() {
  deleteFile(path.resolve(__dirname, './assets'));
  copyFolder(path.resolve(__dirname, './new'), path.resolve(__dirname, './assets'));
  const urls = [...new Array(110)].map((_v, index) => ({
    url: `https://res.wx.qq.com/mpres/htmledition/images/icon/common/emotion_panel/smiley/smiley_${index}.png`,
    name: `wx-${index + 1}`,
  }));

  for (const item of urls) {
    try {
      console.log(`fetching: ${item.name}`);
      await fetchWxEmoji({ url: item.url }, path.resolve(__dirname, `./assets/${item.name}.png`));
    } catch (error) { }
  }
}

update();


src/sprite/index.js
const { ImageInfo, Point, SimpleSprite } = require("./tools");
const fs = require("fs");
const path = require("path");
const images = require("images");
const filenameMap = {};

function generateSprite(targetImagesDirPath) {
  return new Promise(async (resolve, reject) => {
    if (!targetImagesDirPath) {
      return reject(new Error("please input image dir."));
    }
    const imageInfos = fs
      .readdirSync(targetImagesDirPath)
      .filter((filename) => filename !== "sprite.png")
      .filter((filename) => {
        const suffix = filename.substring(filename.indexOf(".")).toLowerCase();
        if (
          suffix !== ".bmp" &&
          suffix !== ".png" &&
          suffix !== ".gif" &&
          suffix !== ".jpg" &&
          suffix !== ".jpeg"
        ) {
          return false;
        }
        return true;
      })
      .map((filename) => {
        const filepath = path.join(targetImagesDirPath, filename);
        filenameMap[filepath] = filename;
        return filepath;
      })
      .filter((filepath) => fs.statSync(filepath).isFile())
      .map((filepath) => new ImageInfo(filepath, images(filepath).size()));

    imageInfos.sort((image1, image2) => image1.area < image2.area);

    const simpleSprite = new SimpleSprite();

    imageInfos.forEach((item) => {
      const size = item.width;
      const region = simpleSprite.getSmallestAreaRegion(size, size, item.path);

      if (region) {
        simpleSprite.regions.push(region);
        simpleSprite.points.push(new Point(region.right, region.top));
        simpleSprite.points.push(new Point(region.left, region.bottom));
        const index = simpleSprite.points.findIndex((point) => {
          return point.x === region.x && point.y === region.y;
        });
        simpleSprite.points.splice(index, 1);
      }
    });

    const sprintImage = images(
      simpleSprite.getTotalWidthOfRegions(),
      simpleSprite.getTotalHeightOfRegions()
    );
    simpleSprite.regions.forEach((region) => {
      sprintImage.draw(images(region.path), region.x, region.y);
    });

    sprintImage.save(path.join(targetImagesDirPath, "../img/sprite.png"));
    // 生成css文件
    let content = `
      .wxem {
        height: 1em;
        width: 1em;
        font-size: 24px;
        background: url(img/sprite.png);
        content: url(img/block.png);
        background-size: 11000%;
        background-repeat: no-repeat;
      }
      \n
    `;

    simpleSprite.regions.forEach((item, index) => {
      const name = (filenameMap[item.path] || "").replace(".png", "");
      let _imgCss = "";

      if (!name) {
        return;
      }

      switch (name) {
        case "wx-new-1":
          _imgCss = "background-position: 91.78% 0px;";
          break;
        case "wx-new-2":
          _imgCss = "background-position: 93.6% 0px;";
          break;
        case "wx-new-3":
          _imgCss = "background-position: 94.48% 0px;";
          break;
        case "wx-new-4":
          _imgCss = "background-position: 95.4% 0px;";
          break;
        case "wx-new-5":
          _imgCss = "background-position: 96.36% 0px;";
          break;
        case "wx-new-6":
          _imgCss = "background-position: 97.24% 0px;";
          break;
        case "wx-new-7":
          _imgCss = "background-position: 98.2% 0px;";
          break;
        case "wx-new-8":
          _imgCss = "background-position: 99.1% 0px;";
          break;
        case "wx-new-9":
          _imgCss = "background-position: 99.99% 0px;";
          break;
        case "wx-new-10":
          _imgCss = "background-position: 92.7% 0px;";
          break;
        default:
          _imgCss = `background-position: ${index * 0.8735294117647059}% 0px;`;
          break;
      }

      _imgCss = `
        .wxem-${name} {
          ${_imgCss}
        }\n
      `;

      content = content + _imgCss;
    });

    fs.writeFileSync(
      path.join(targetImagesDirPath, "../css/index.css"),
      content
    );

    setTimeout(() => {
      console.info(
        `generate sprite image success,it has been saved in ${targetImagesDirPath}/sprite.png`
      );
      console.info(
        `generate css success,it has been saved in ${targetImagesDirPath}/sprite-css.txt`
      );
      resolve();
    }, 1000);
  });
}

module.exports = {
  generateSprite,
};


src/sprite/tools.js
class ImageInfo {
  constructor(path, size) {
    this.path = path;
    this.width = size.width;
    this.height = size.height;
    this.area = this.width * this.height;
  }
}

class Point {
  constructor(x, y) {
    this.x = x;
    this.y = y;
  }
}

class Region {
  constructor(x, y, width, height, path) {
    this.path = path;
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
    this.left = x;
    this.top = y;
    this.right = this.left + width;
    this.bottom = this.top + height;
    this.centerX = (this.left + this.right) / 2;
    this.centerY = (this.top + this.bottom) / 2;
  }

  isCollided(region) {
    var collided =
      Math.abs(this.centerX - region.centerX) < (this.width + region.width) / 2 &&
      Math.abs(this.centerY - region.centerY) < (this.height + region.height) / 2;
    return collided;
  }
}

class SimpleSprite {
  constructor() {
    this.points = [];
    this.regions = [];
    this.initPoints();
  }

  initPoints() {
    const startPoint = new Point(0, 0);
    this.points.push(startPoint);
  }

  getPossibleRegions(width, height, path) {
    const regionList = [];

    this.points.forEach(point => {
      const region_new = new Region(point.x, point.y, width, height, path);
      const len = this.regions.length;
      if (len === 0) {
        regionList.push(region_new);
      } else {
        let isAllNotCollided = true;
        for (let i = 0; i < len; i++) {
          if (this.regions[i].isCollided(region_new)) {
            isAllNotCollided = false;
          }
        }
        // eslint-disable-next-line @typescript-eslint/no-unused-expressions
        isAllNotCollided ? regionList.push(region_new) : null;
      }
    });
    return regionList;
  }

  getSmallestAreaRegion(width, height, path) {
    const regionList = this.getPossibleRegions(width, height, path);

    let minarea = Number.MAX_VALUE;
    let minRegion;
    regionList.forEach(item => {
      const area = this.getTotalAreaOfRegions(item);
      if (minarea > area) {
        minarea = area;
        minRegion = item;
      }
    });
    return minRegion;
  }

  getTotalAreaOfRegions(region) {
    const w = this.getTotalWidthOfRegions(region);
    const h = this.getTotalHeightOfRegions(region);
    return w * h;
  }

  getTotalWidthOfRegions(region) {
    let _totalWidth = region ? region.right : 0;
    this.regions.forEach(item => {
      _totalWidth = Math.max(_totalWidth, item.right);
    });
    return _totalWidth;
  }

  getTotalHeightOfRegions(region) {
    let _totalHeight = region ? region.bottom : 0;
    this.regions.forEach(item => {
      _totalHeight = Math.max(_totalHeight, item.bottom);
    });
    return _totalHeight;
  }
}

module.exports = {
  ImageInfo,
  Point,
  Region,
  SimpleSprite,
};


src/emojis.wx.js
/**
 * http://www.360doc.com/content/16/0116/21/2697712_528463222.shtml
 * https://developers.weixin.qq.com/miniprogram/dev/extended/component-plus/emoji.html
 */

const emojis_wx = [
  {
    emoji: '[微笑]',
    name: '微笑',
    em_code: '/::)',
    code: 'wx-1',
  },
  {
    emoji: '[撇嘴]',
    name: '撇嘴',
    em_code: '/::~',
    code: 'wx-2',
  },
  {
    emoji: '[色]',
    name: '色',
    em_code: '/::B',
    code: 'wx-3',
  },
  {
    emoji: '[发呆]',
    name: '发呆',
    em_code: '/::|',
    code: 'wx-4',
  },
  {
    emoji: '[得意]',
    name: '得意',
    em_code: '/:8-)',
    code: 'wx-5',
  },
  {
    emoji: '[流泪]',
    name: '流泪',
    em_code: '/::<',
    code: 'wx-6',
  },
  {
    emoji: '[害羞]',
    name: '害羞',
    em_code: '/::$',
    code: 'wx-7',
  },
  {
    emoji: '[闭嘴]',
    name: '闭嘴',
    em_code: '/::X',
    code: 'wx-8',
  },
  {
    emoji: '[睡]',
    name: '睡',
    em_code: '/::Z',
    code: 'wx-9',
  },
  {
    emoji: '[大哭]',
    name: '大哭',
    em_code: '/::’(',
    code: 'wx-10',
  },
  {
    emoji: '[尴尬]',
    name: '尴尬',
    em_code: '/::-|',
    code: 'wx-11',
  },
  {
    emoji: '[发怒]',
    name: '发怒',
    em_code: '/::@',
    code: 'wx-12',
  },
  {
    emoji: '[调皮]',
    name: '调皮',
    em_code: '/::P',
    code: 'wx-13',
  },
  {
    emoji: '[呲牙]',
    name: '呲牙',
    em_code: '/::D',
    code: 'wx-14',
  },
  {
    emoji: '[惊讶]',
    name: '惊讶',
    em_code: '/::O',
    code: 'wx-15',
  },
  // line2
  {
    emoji: '[难过]',
    name: '难过',
    em_code: '/::(',
    code: 'wx-16',
  },
  {
    emoji: '[酷]',
    name: '酷',
    em_code: '/::+',
    code: 'wx-17',
  },
  {
    emoji: '[囧]',
    name: '囧',
    em_code: '/:–b',
    code: 'wx-18',
  },
  {
    emoji: '[抓狂]',
    name: '抓狂',
    em_code: '/::Q',
    code: 'wx-19',
  },
  {
    emoji: '[吐]',
    name: '吐',
    em_code: '/::T',
    code: 'wx-20',
  },
  {
    emoji: '[偷笑]',
    name: '偷笑',
    em_code: '/:,@P',
    code: 'wx-21',
  },
  {
    emoji: '[愉快]',
    name: '愉快',
    em_code: '/:,@-D',
    code: 'wx-22',
  },
  {
    emoji: '[白眼]',
    name: '白眼',
    em_code: '/::d',
    code: 'wx-23',
  },
  {
    emoji: '[傲慢]',
    name: '傲慢',
    em_code: '/:,@o',
    code: 'wx-24',
  },
  {
    emoji: '[饥饿]',
    name: '饥饿',
    em_code: '/::g',
    code: 'wx-25',
  },
  {
    emoji: '[困]',
    name: '困',
    em_code: '/:|-)',
    code: 'wx-26',
  },
  {
    emoji: '[惊恐]',
    name: '惊恐',
    em_code: '/::!',
    code: 'wx-27',
  },
  {
    emoji: '[流汗]',
    name: '流汗',
    em_code: '/::L',
    code: 'wx-28',
  },
  {
    emoji: '[憨笑]',
    name: '憨笑',
    em_code: '/::>',
    code: 'wx-29',
  },
  {
    emoji: '[悠闲]',
    name: '悠闲',
    em_code: '/::,@',
    code: 'wx-30',
  },
  // line3
  {
    emoji: '[奋斗]',
    name: '奋斗',
    em_code: '/:,@f',
    code: 'wx-31',
  },
  {
    emoji: '[咒骂]',
    name: '咒骂',
    em_code: '/::-S',
    code: 'wx-32',
  },
  {
    emoji: '[疑问]',
    name: '疑问',
    em_code: '/:?',
    code: 'wx-33',
  },
  {
    emoji: '[嘘]',
    name: '嘘',
    em_code: '/:,@x',
    code: 'wx-34',
  },
  {
    emoji: '[晕]',
    name: '晕',
    em_code: '/:,@@',
    code: 'wx-35',
  },
  {
    emoji: '[疯了]',
    name: '疯了',
    em_code: '/::8',
    code: 'wx-36',
  },
  {
    emoji: '[衰]',
    name: '衰',
    em_code: '/:,@!',
    code: 'wx-37',
  },
  {
    emoji: '[骷髅]',
    name: '骷髅',
    em_code: '/:!!!',
    code: 'wx-38',
  },
  {
    emoji: '[敲打]',
    name: '敲打',
    em_code: '/:xx',
    code: 'wx-39',
  },
  {
    emoji: '[再见]',
    name: '再见',
    em_code: '/:bye',
    code: 'wx-40',
  },
  {
    emoji: '[擦汗]',
    name: '擦汗',
    em_code: '/:wipe',
    code: 'wx-41',
  },
  {
    emoji: '[抠鼻]',
    name: '抠鼻',
    em_code: '/:dig',
    code: 'wx-42',
  },
  {
    emoji: '[鼓掌]',
    name: '鼓掌',
    em_code: '/:handclap',
    code: 'wx-43',
  },
  {
    emoji: '[糗大了]',
    name: '糗大了',
    em_code: '/:&-(',
    code: 'wx-44',
  },
  {
    emoji: '[坏笑]',
    name: '坏笑',
    em_code: '/:B-)',
    code: 'wx-45',
  },
  // line4
  {
    emoji: '[左哼哼]',
    name: '左哼哼',
    em_code: '/:<@',
    code: 'wx-46',
  },
  {
    emoji: '[右哼哼]',
    name: '右哼哼',
    em_code: '/:@>',
    code: 'wx-47',
  },
  {
    emoji: '[哈欠]',
    name: '哈欠',
    em_code: '/::-O',
    code: 'wx-48',
  },
  {
    emoji: '[鄙视]',
    name: '鄙视',
    em_code: '/:>-|',
    code: 'wx-49',
  },
  {
    emoji: '[委屈]',
    name: '委屈',
    em_code: '/:P-(',
    code: 'wx-50',
  },
  {
    emoji: '[快哭了]',
    name: '快哭了',
    em_code: '/::’|',
    code: 'wx-51',
  },
  {
    emoji: '[阴险]',
    name: '阴险',
    em_code: '/:X-)',
    code: 'wx-52',
  },
  {
    emoji: '[亲亲]',
    name: '亲亲',
    em_code: '/::*',
    code: 'wx-53',
  },
  {
    emoji: '[吓]',
    name: '吓',
    em_code: '/:@x',
    code: 'wx-54',
  },
  {
    emoji: '[可怜]',
    name: '可怜',
    em_code: '/:8*',
    code: 'wx-55',
  },
  {
    emoji: '[菜刀]',
    name: '菜刀',
    em_code: '/:pd',
    code: 'wx-56',
  },
  {
    emoji: '[西瓜]',
    name: '西瓜',
    em_code: '/:<W>',
    code: 'wx-57',
  },
  {
    emoji: '[啤酒]',
    name: '啤酒',
    em_code: '/:beer',
    code: 'wx-58',
  },
  {
    emoji: '[篮球]',
    name: '篮球',
    em_code: '/:basketb',
    code: 'wx-59',
  },
  {
    emoji: '[乒乓]',
    name: '乒乓',
    em_code: '/:oo',
    code: 'wx-60',
  },
  // line5
  {
    emoji: '[咖啡]',
    name: '咖啡',
    em_code: '/:coffee',
    code: 'wx-61',
  },
  {
    emoji: '[饭]',
    name: '饭',
    em_code: '/:eat',
    code: 'wx-62',
  },
  {
    emoji: '[猪头]',
    name: '猪头',
    em_code: '/:pig',
    code: 'wx-63',
  },
  {
    emoji: '[玫瑰]',
    name: '玫瑰',
    em_code: '/:rose',
    code: 'wx-64',
  },
  {
    emoji: '[凋谢]',
    name: '凋谢',
    em_code: '/:fade',
    code: 'wx-65',
  },
  {
    emoji: '[嘴唇]',
    name: '嘴唇',
    em_code: '/:showlove',
    code: 'wx-66',
  },
  {
    emoji: '[爱心]',
    name: '爱心',
    em_code: '/:heart',
    code: 'wx-67',
  },
  {
    emoji: '[心碎]',
    name: '心碎',
    em_code: '/:break',
    code: 'wx-68',
  },
  {
    emoji: '[蛋糕]',
    name: '蛋糕',
    em_code: '/:cake',
    code: 'wx-69',
  },
  {
    emoji: '[闪电]',
    name: '闪电',
    em_code: '/:li',
    code: 'wx-70',
  },
  {
    emoji: '[炸弹]',
    name: '炸弹',
    em_code: '/:bome',
    code: 'wx-71',
  },
  {
    emoji: '[刀]',
    name: '刀',
    em_code: '/:kn',
    code: 'wx-72',
  },
  {
    emoji: '[足球]',
    name: '足球',
    em_code: '/:footb',
    code: 'wx-73',
  },
  {
    emoji: '[瓢虫]',
    name: '瓢虫',
    em_code: '/:ladybug',
    code: 'wx-74',
  },
  {
    emoji: '[便便]',
    name: '便便',
    em_code: '/:shit',
    code: 'wx-75',
  },
  // line6
  {
    emoji: '[月亮]',
    name: '月亮',
    em_code: '/:moon',
    code: 'wx-76',
  },
  {
    emoji: '[太阳]',
    name: '太阳',
    em_code: '/:sun',
    code: 'wx-77',
  },
  {
    emoji: '[礼物]',
    name: '礼物',
    em_code: '/:gift',
    code: 'wx-78',
  },
  {
    emoji: '[拥抱]',
    name: '拥抱',
    em_code: '/:hug',
    code: 'wx-79',
  },
  {
    emoji: '[强]',
    name: '强',
    em_code: '/:strong',
    code: 'wx-80',
  },
  {
    emoji: '[弱]',
    name: '弱',
    em_code: '/:weak',
    code: 'wx-81',
  },
  {
    emoji: '[握手]',
    name: '握手',
    em_code: '/:share',
    code: 'wx-82',
  },
  {
    emoji: '[胜利]',
    name: '胜利',
    em_code: '/:v',
    code: 'wx-83',
  },
  {
    emoji: '[抱拳]',
    name: '抱拳',
    em_code: '/:@)',
    code: 'wx-84',
  },
  {
    emoji: '[勾引]',
    name: '勾引',
    em_code: '/:jj',
    code: 'wx-85',
  },
  {
    emoji: '[拳头]',
    name: '拳头',
    em_code: '/:@@',
    code: 'wx-86',
  },
  {
    emoji: '[差劲]',
    name: '差劲',
    em_code: '/:bad',
    code: 'wx-87',
  },
  {
    emoji: '[爱你]',
    name: '爱你',
    em_code: '/:lvu',
    code: 'wx-88',
  },
  {
    emoji: '[NO]',
    name: 'NO',
    em_code: '/:no',
    code: 'wx-89',
  },
  {
    emoji: '[OK]',
    name: 'OK',
    em_code: '/:ok',
    code: 'wx-90',
  },
  // line7
  {
    emoji: '[爱情]',
    name: '爱情',
    em_code: '/:love',
    code: 'wx-91',
  },
  {
    emoji: '[飞吻]',
    name: '飞吻',
    em_code: '/:<L>',
    code: 'wx-92',
  },
  {
    emoji: '[跳跳]',
    name: '跳跳',
    em_code: '/:jump',
    code: 'wx-93',
  },
  {
    emoji: '[发抖]',
    name: '发抖',
    em_code: '/:shake',
    code: 'wx-94',
  },
  {
    emoji: '[怄火]',
    name: '怄火',
    em_code: '/:<o>',
    code: 'wx-95',
  },
  {
    emoji: '[转圈]',
    name: '转圈',
    em_code: '/:circle',
    code: 'wx-96',
  },
  {
    emoji: '[磕头]',
    name: '磕头',
    em_code: '/:kotow',
    code: 'wx-97',
  },
  {
    emoji: '[回头]',
    name: '回头',
    em_code: '/:turn',
    code: 'wx-98',
  },
  {
    emoji: '[跳绳]',
    name: '跳绳',
    em_code: '/:skip',
    code: 'wx-99',
  },
  {
    emoji: '[投降]',
    name: '投降',
    em_code: '/:oy',
    code: 'wx-100',
  },
  {
    emoji: '[激动]',
    name: '激动',
    em_code: '/:#-0',
    code: 'wx-101',
  },
  {
    emoji: '[乱舞]',
    name: '乱舞',
    em_code: '/:hiphot',
    code: 'wx-102',
  },
  {
    emoji: '[献吻]',
    name: '献吻',
    em_code: '/:kiss',
    code: 'wx-103',
  },
  {
    emoji: '[左太极]',
    name: '左太极',
    em_code: '/:<&',
    code: 'wx-104',
  },
  {
    emoji: '[右太极]',
    name: '右太极',
    em_code: '/:&>',
    code: 'wx-105',
  },
  // 2020微信新加的表情
  {
    emoji: '[吃瓜]',
    name: '吃瓜',
    em_code: '',
    code: 'wx-new-1',
  },
  {
    emoji: '[Emm]',
    name: 'Emm',
    em_code: '',
    code: 'wx-new-2',
  },
  {
    emoji: '[哇]',
    name: '哇',
    em_code: '',
    code: 'wx-new-3',
  },
  {
    emoji: '[社会社会]',
    name: '社会社会',
    em_code: '',
    code: 'wx-new-4',
  },

  {
    emoji: '[汗]',
    name: '汗',
    em_code: '',
    code: 'wx-new-5',
  },
  {
    emoji: '[加油]',
    name: '加油',
    em_code: '',
    code: 'wx-new-6',
  },
  {
    emoji: '[天啊]',
    name: '天啊',
    em_code: '',
    code: 'wx-new-7',
  },
  {
    emoji: '[好的]',
    name: '好的',
    em_code: '',
    code: 'wx-new-8',
  },
  {
    emoji: '[打脸]',
    name: '打脸',
    em_code: '',
    code: 'wx-new-9',
  },
  {
    emoji: '[旺柴]',
    name: '旺柴',
    em_code: '',
    code: 'wx-new-10',
  },
];

export default emojis_wx;
